modules/node_modules/@ncigdc/dux/bannerNotification.js

```diff
@@ -0,0 +1,62 @@
+/* @flow */
+
+import React from 'react';
+
+import DismissibleBanner from '@ncigdc/components/DismissibleBanner';
+
+import { fetchApi } from '@ncigdc/utils/ajax';
+
+const NOTIFICATION_SUCCESS = 'NOTIFICATION_SUCCESS';
+const NOTIFICATION_DISMISS = 'NOTIFICATION_DISMISS';
+
+
+export function fetchNotifications() {
+  return async (dispatch) => {
+    let { data } = await fetchApi(
+      'notifications',
+      { headers: { 'Content-Type': 'application/json' } }
+    );
+    
+    dispatch({
+      type: NOTIFICATION_SUCCESS,
+      payload: data
+    });
+  };
+}
+
+export function dismissNotification(notificationID) {
+  return {
+    type: NOTIFICATION_DISMISS,
+    payload: { id: notificationID }
+  };
+}
+
+let initialState = [{
+  components:['PORTAL'],
+  level:'INFO',
+  id: 0,
+  dismissible: true,
+  message: <DismissibleBanner />,
+}];
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case NOTIFICATION_SUCCESS:
+      return [
+        ...action.payload
```

Comment:
can you wrap this in a null check? if bad `data` comes back from the server this crashes

```
...(action.payload || []).filter
```